parlai/core/teachers.py

```diff
@@ -675,6 +705,17 @@ def __init__(self, opt, data_loader=None, cands=None, shared=None, **kwargs):
         else:
             self.image_loader = ImageLoader(opt)
             self.data = []
+
+            if 'datafile' not in opt:
+                raise KeyError(
```

Comment:
nit: maybe abstract out this error message instead of repeating it in 3 different places?